org.eclipse.jdt.debug/model/org/eclipse/jdt/internal/debug/core/model/JDIDebugTarget.java

```diff
@@ -1340,6 +1344,13 @@ protected void resume(boolean fireNotification) throws DebugException {
 			resumeThreads();
 			VirtualMachine vm = getVM();
 			if (vm != null) {
+				if (pendingResumeRequests) {
+					while (suspendCount > 1) {
```

Comment:
`suspendCount`, once set from zero to `> 1`, will stay always `>= 1` after this while loop.
Why not check `while (suspendCount > 0)`?

Reply:
I've fixed that in my last pushed commit.